src/SFA.DAS.EmployerIncentives.Api/Controllers/WithdrawalCommandController.cs

```diff
@@ -31,10 +44,35 @@ public async Task<IActionResult> WithdrawalIncentiveApplication([FromBody] Withd
                         request.ServiceRequest.TaskCreatedDate??DateTime.UtcNow));
                 return Accepted();
             }
+            else if(request.WithdrawalType == WithdrawalType.Compliance)
+            {
+                await SendCommandAsync(
+                    new ComplianceWithdrawalCommand(
+                        request.AccountLegalEntityId,
+                        request.ULN,
+                        request.ServiceRequest.TaskId,
+                        request.ServiceRequest.DecisionReference,
+                        request.ServiceRequest.TaskCreatedDate ?? DateTime.UtcNow));
+                return Accepted();
+            }
             else
             {
                 return BadRequest();
             }
         }       
+
+        private async Task<bool> HasPayments(long accountLegalEntityId, long uln)
+        {
+            var incentiveModel = await _queryRepository.Get((a =>
```

Comment:
a job for a query-handler (which can be used with Value Types) ?

Reply:
Maybe but Is that adding extra layers unnecessarily? The IQueryRepository<T> as implemented by  IApprenticeshipIncentiveQueryRepository supports read only querying so what value would a handler add?

Reply:
I can update to fit in with elsewhere if required.

Reply:
Updated